solutions/src/main/java/ru/sberbank/school/task10/ScalableThreadPool.java

```diff
@@ -0,0 +1,130 @@
+package ru.sberbank.school.task10;
+
+import lombok.NonNull;
+
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Future;
+import java.util.concurrent.FutureTask;
+
+public class ScalableThreadPool implements ThreadPool {
+    private final Queue<FutureTask> tasks = new LinkedList<>();
+    private PoolWorker[] threads;
+    private int minSizePool;
+
+    public ScalableThreadPool(int min, int max) {
+        if (min <= 0 || max <= 0) {
+            throw new IllegalArgumentException("Amount of threads must be more than zero");
+        }
+        if (min > max) {
+            throw new IllegalArgumentException("Minimum amount of threads must be less than maximum");
+        }
+
+        threads = new PoolWorker[max];
+
+        for (int i = 0; i < threads.length; i++) {
+            threads[i] = new PoolWorker("ThreadPoolWorker-" + i);
+        }
+
+        minSizePool = min;
+    }
+
+    @Override
+    public void start() {
+        for (int i = 0; i < minSizePool; i++) {
+            threads[i].start();
+        }
+    }
+
+    @Override
+    public void stopNow() {
+        for (PoolWorker thread : threads) {
+            thread.interrupt();
+        }
+        synchronized (tasks) {
+            tasks.clear();
+        }
+
+        threads = null;
+    }
+
+    @Override
+    public void execute(@NonNull Runnable runnable) {
+        synchronized (tasks) {
+            tasks.add(new FutureTask<>(runnable, true));
+
+            correctSizeThreadPool();
+
+            tasks.notifyAll();
+        }
+    }
+
+    @Override
+    public <T> Future<T> execute(@NonNull Callable<T> callable) {
+        synchronized (tasks) {
+            FutureTask<T> future = new FutureTask<>(callable);
+
+            tasks.add(future);
+
+            correctSizeThreadPool();
+
+            tasks.notifyAll();
+
+            return future;
+        }
+    }
+
+    /**
```

Comment:
First sentence of Javadoc is missing an ending period.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-825542165>*